### DISCUSSION TITLE
Potential mod support?

### DISCUSSION THREAD

**Discussion by IAmTheTable** [Pumpkin-MC/Pumpkin#99] [Ideas]
Hello development team, this is an amazing idea brought to life! One additional feature/version I’m sure the community would like to see, is mod support/forge revisions. I myself am interested in such fork and if there is please educate me.

**Comment by Bryntet** [comment]
Mod support will only ever be on the table through a custom pumpkin plugin that translates the mod calls. 

Pumpkin is still far from feature parity with vanilla, and I think we probably won't ever support mods officially, as the official Minecraft is work enough

**Comment by Snowiiii** [comment]
I saw there are certain Packets send and received by Forge Clients. Reference: https://wiki.vg/Minecraft_Forge_Handshake. Adding support for Packets is not a Problem. We currently not planning to translate Java API calls to Rust. Keep in mind that The overall Plugin API will have more Features than the current limited Bukkit API.

**Reply by Bryntet** [reply]
IMO, we should add support for plugins extending new custom packet support.

We should not support specific extra packet things, it will become an unmaintainable mountain of development time pit

**Reply by Snowiiii** [reply]
i mean. I also wanted to add support for custom Proxies (e.g. Velocity)

**Reply by Bryntet** [reply]
I see a big difference in supporting a proxy protocol, that's quite statically defined, compared to a constantly changing modding thing

**Comment by KevinCCucumber** [comment]
If I may throw my hat into the ring: Why not start with fabric? Afaik its supposed to me much more lightweight than forge.
I dont know about specific handshakes though.

**Reply by NGSpace** [reply]
Fabric encourages mods to interact with Minecraft's code directly which will be a much harder translation, not only will you have to translate the calls to API functions, you'd also need to translate the calls to every function that exists in minecraft's code.

And that's ignoring how much Fabric relies on mixins which IDEK how they would even be implemented, as far as I know there is no easy way to replace code at runtime in rust, let alone code that has to be translated from Java (once again, with the Fabric API and Minecraft Function calls translated)


\- A Fabric mod dev


I'd suggest Pumpkin carve their own path and make their own plugin API.

**Comment by vosta09** [comment]
....
